pkg/apisix/nonexistentclient.go

```diff
@@ -132,15 +160,19 @@ type dummyCache struct{}
 
 var _ cache.Cache = &dummyCache{}
 
-func (c *dummyCache) InsertRoute(_ *v1.Route) error              { return nil }
-func (c *dummyCache) InsertSSL(_ *v1.Ssl) error                  { return nil }
-func (c *dummyCache) InsertUpstream(_ *v1.Upstream) error        { return nil }
-func (c *dummyCache) GetRoute(_ string) (*v1.Route, error)       { return nil, cache.ErrNotFound }
-func (c *dummyCache) GetSSL(_ string) (*v1.Ssl, error)           { return nil, cache.ErrNotFound }
-func (c *dummyCache) GetUpstream(_ string) (*v1.Upstream, error) { return nil, cache.ErrNotFound }
-func (c *dummyCache) ListRoutes() ([]*v1.Route, error)           { return nil, nil }
-func (c *dummyCache) ListSSL() ([]*v1.Ssl, error)                { return nil, nil }
-func (c *dummyCache) ListUpstreams() ([]*v1.Upstream, error)     { return nil, nil }
-func (c *dummyCache) DeleteRoute(_ *v1.Route) error              { return nil }
-func (c *dummyCache) DeleteSSL(_ *v1.Ssl) error                  { return nil }
-func (c *dummyCache) DeleteUpstream(_ *v1.Upstream) error        { return nil }
+func (c *dummyCache) InsertRoute(_ *v1.Route) error                    { return nil }
+func (c *dummyCache) InsertSSL(_ *v1.Ssl) error                        { return nil }
+func (c *dummyCache) InsertUpstream(_ *v1.Upstream) error              { return nil }
+func (c *dummyCache) InsertStreamRoute(_ *v1.StreamRoute) error        { return nil }
+func (c *dummyCache) GetRoute(_ string) (*v1.Route, error)             { return nil, cache.ErrNotFound }
+func (c *dummyCache) GetSSL(_ string) (*v1.Ssl, error)                 { return nil, cache.ErrNotFound }
+func (c *dummyCache) GetUpstream(_ string) (*v1.Upstream, error)       { return nil, cache.ErrNotFound }
+func (c *dummyCache) GetStreamRoute(_ string) (*v1.StreamRoute, error) { return nil, cache.ErrNotFound }
+func (c *dummyCache) ListRoutes() ([]*v1.Route, error)                 { return nil, nil }
+func (c *dummyCache) ListSSL() ([]*v1.Ssl, error)                      { return nil, nil }
+func (c *dummyCache) ListUpstreams() ([]*v1.Upstream, error)           { return nil, nil }
+func (c *dummyCache) ListStreamRoutes() ([]*v1.StreamRoute, error)     { return nil, nil }
+func (c *dummyCache) DeleteRoute(_ *v1.Route) error                    { return nil }
+func (c *dummyCache) DeleteSSL(_ *v1.Ssl) error                        { return nil }
+func (c *dummyCache) DeleteUpstream(_ *v1.Upstream) error              { return nil }
+func (c *dummyCache) DeleteStreamRoute(_ *v1.StreamRoute) error        { return nil }
```

Comment:
Can these methods be refactored to be together, 
such as, dummyCache.Insert(), dummyCache.List()

Reply:
OK, will change it.